CHANGES.md

```diff
@@ -1,5 +1,124 @@
 # Changes
 
+## stable-2.12.0
+
+This release introduces route-based policy to Linkerd, allowing users to define
+and enforce authorization policies based on HTTP routes in a fully zero-trust
+way. These policies are built on Linkerd's strong workload identities, secured
+by mutual TLS, and configured using types from Kubernetes's [Gateway API](https://gateway-api.sigs.k8s.io/).
+
+The 2.12 release also introduces access logging, a long-awaited feature that
+optionally allows Linkerd to produce Apache-style request logs, optional
+support for `iptables-nft`, and a host of other improvements and performance
+enhancements.
+
+**Upgrade notes**: Please see the [upgrade instructions](upgrade-2120). <!-- TODO fix link -->
+
+* Proxy
+  * Added a `config.linkerd.io/shutdown-grace-period` annotation to configure
+   the proxy's maximum grace period for graceful shutdown
+  * Added a new `iptables-nft` mode for `proxy-init`
+  * Fixed an issue where the proxy-injector would break when using
+    `nodeAffinity` values for the control plane
+
+* Control Plane
+  * Added support for per-route policy by supporting AuthorizationPolicy
+    resources which target HttpRoute resources
+  * Removed kube-system exclusions from watchers to fix service discovery
+    for workloads in the kube-system namespace (thanks @JacobHenner)
+  * Updated healthcheck to ignore `Terminated` state for pods (thanks
+    @AgrimPrasad!)
+  * Updated the default policy controller log level to `info`; the controller
+    will now emit INFO level logs for some of its dependencies
```

Comment:
This is just my personal opinion, it would probably be best to hear what others think, but this doesn't sound like something worth mentioning necessarily. While it is user facing, I'm not sure if it's something people should be aware of when the release drops.

Reply:
I'd personally lean toward including all user-facing stuff, but it's not a strong yes – happy to defer to others.